### ISSUE TITLE
oclint invokes clang with incorrect search path for C++ headers on macOS 10.11

### ISSUE THREAD

**Issue by brennie** [oclint/oclint#358]
I have installed `oclint` via Homebrew and have the following source file:

```cpp
/* src/main.cpp */
#include <iostream>
int main() {
    std::cout << "Hello, world" << std::endl;
    return 0;
}
```

When I run `oclint src/main.cpp` I get the following output:
```
rror while trying to load a compilation database:
Could not auto-detect compilation database for file "src/main.cpp"
No compilation database found in /Users/barret/.workspace/src/github.com/brennie/displayctrl/src or any parent directory
json-compilation-database: Error while opening JSON database: No such file or directory
Running without flags.

Compiler Errors:
(please be aware that these errors will prevent OCLint from analyzing this source code)

/Users/barret/.workspace/src/github.com/brennie/displayctrl/src/main.cpp:1:10: 'iostream' file not found


OCLint Report

Summary: TotalFiles=0 FilesWithViolations=0 P1=0 P2=0 P3=0


[OCLint (http://oclint.org) v0.10.3]
```

I ran again with `-- -v` to see the `clang` invocation and noticed that it is searching the wrong directories:

```
[snip]
ignoring nonexistent directory "/usr/local/Cellar/oclint/0.10.3/bin/../include/c++/v1"
ignoring nonexistent directory "/usr/include/c++/v1"
#include "..." search starts here:
#include <...> search starts here:
 /usr/local/include
 /usr/local/Cellar/oclint/0.10.3/bin/../lib/clang/3.8.0/include
 /usr/include
 /System/Library/Frameworks (framework directory)
 /Library/Frameworks (framework directory)
[snip]
```

The output of `clang -v src/main.cpp` follows:
```
[snip]
ignoring nonexistent directory "/usr/include/c++/v1"
#include "..." search starts here:
#include <...> search starts here:
 /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/../include/c++/v1
 /usr/local/include
 /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/../lib/clang/7.3.0/include
 /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include
 /usr/include
 /System/Library/Frameworks (framework directory)
 /Library/Frameworks (framework directory)
End of search list.
[snip]
```

`oclint` should be using the correct C++ header directory (`Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/../include/c++/v1`) on macOS, but it is not.

**Comment by ryuichis** [comment]
My concern is `ignoring nonexistent directory "/usr/local/Cellar/oclint/0.10.3/bin/../include/c++/v1"`, could you check if that folder exists on your machine?

**Comment by ryuichis** [comment]
Ah, I probably know the reason. Thanks for bringing this up. Question: did you install from our homebrew https://github.com/oclint/homebrew-formulae or homebrew-cask https://github.com/caskroom/homebrew-cask/blob/master/Casks/oclint.rb?

**Comment by brennie** [comment]
I've installed it from `oclint/homebrew-formulae`: 

```
$ brew info oclint
oclint/formulae/oclint: stable 0.10.3, HEAD
OCLint static code analysis tool for C, C++, and Objective-C
http://oclint.org
/usr/local/Cellar/oclint/0.10.3 (175 files, 51.0M) *
  Built from source on 2016-07-21 at 22:07:30
From: https://github.com/oclint/homebrew-formulae/blob/master/Formula/oclint.rb
```

Also, that folder does not exist:

```
$ ls /usr/local/Cellar/oclint/0.10.3/
INSTALL_RECEIPT.json  LICENSE  bin/  lib/
```

**Comment by ryuichis** [comment]
@brennie it has been addressed at our tap. The cask tap will be updated accordingly when we release our next version. Thanks for bringing this up to our attention.
